### PR TITLE
Change menu icon to Material Component.

### PR DESCRIPTION
**Pull request by BugsNash**
Part of #369.

Also adds the libraries required to use Material Components and Material Design icons.

Style tweaked to match Material Design [guidelines](https://material.io/design/iconography/system-icons.html#color).

Review guidelines:
The changes inside third_party can be ignored as that is me adding the [material-design-icons](https://github.com/google/material-design-icons) library as is and the minified version of the material-comonents-web [css](https://unpkg.com/material-components-web@latest/dist/material-components-web.min.css) and [js](https://unpkg.com/material-components-web@latest/dist/material-components-web.min.js) files retrieved from unpkg, along with licence from github [repo](https://github.com/material-components/material-components-web).
The first 4 files in the list of files changed should be reviewed (these are all the files changed outside of third_party). List list of files is:
- build.py
- css/app.css
- index.html
- js/controllers/window.js

Note that with this patch the menu icon will sometimes render as a flash of the text literal 'menu' during load due to a chrome [intervention](https://www.chromestatus.com/feature/5636954674692096). This seems to be working as intended for font fallbacks and Material Component icons. Not sure why it would be slow to load the font sometimes since it's a file local to the extension, and I can't reliably reproduce it.

### CODE REVIEW

**Comment by BugsNash** [comment]
You can hold off on review, this patch currently triggers a crash in the build script (caused by a bug in the build script but probably means there is some new build error/warning that needs fixing here).

**Comment by E-T-** [comment]
Request: Can this be split into multiple patches? I'm not sure if it makes sense, so feel free to push back, but I wonder if you can do multiple patches:

1. one patch to add the actual MDC library
1. one patch to use the MDC library and style icons correctly for one icon

and potentially just do subsets of icons per patch?

I dunno if that makes a lot of sense as a way of breaking this up, it's just quite hard to review right now in my opinion.

**Comment by BugsNash** [comment]
So the aforementioned crash has been fixed now in other patches, but this patch still doesn't work as is (it does in dev version but once compiled it's missing some of the files it needs). I need to update the build script to include the icon font and css in the compiled version.

**Comment by E-T-** [comment]
Re: WOFFs and CSS: At the moment I'd have thought you'd want to leave the entire library in the repo and only copy anything you need into the compiled version of the app via the build script. (So option 3, although I could see you'd want option 4 as it is simpler).

Re: JS: I thought you'd want to just copy the file into the compiled version. Or do you end up without the definitions of certain classes, and you can't compile your own JS as a result?

**Comment by BugsNash** [comment]
For woff and CSS option 3 is actually simpler to implement than option 4 cause the build script doesn't currently have any recursive copying. I thought it might be good to include the redundancies cause.... Whatever reasons they exist in the first place. Something something. Happy to go with option 3.


For js I can compile fine without including the library in the closure unit (for this unit we're only using simple optimisations) and now that I type that out loud it is clear that the closure compiler would add no value to the already minified library. So yes to copying.

Thanks :)

File: js/controllers/window.js [hunk 1]
```diff
@@ -28,8 +28,8 @@ function WindowController(editor, settings, analytics, tabs) {
   }
 }
 
-/** Initializes UI interaction animatins. */
-WindowController.prototype.initAnimations_ = function() {
+/** Initializes UI elements and animations. */
+WindowController.prototype.initUI_ = function() {
```
Comment (E-T-) [inline js/controllers/window.js]: I feel like the intent of the _ is to make this a private method. Can you please add the @private annotation to this patch please? :)

File: js/controllers/window.js [hunk 2]
```diff
@@ -27,6 +28,12 @@ function WindowController(editor, settings, analytics, tabs) {
   }
 }
 
+/** Initializes UI elements and animations. */
```
Comment (E-T-) [inline js/controllers/window.js]: Actually, I feel like you might as well inline this call. It's not useful to break the linearity of your code for a few lines, in my opinion.

Comment (E-T-) [reply js/controllers/window.js]: I would inline it for now, and when it does get too long we can consider breaking it out. Right now, I think having a single comment would be sufficient (e.g. something like `// Initialize Material UI components.`).

File: js/controllers/window.js (hunk 1 above)
Comment (BugsNash) [reply js/controllers/window.js]: Ack.

File: js/controllers/window.js (hunk 2 above)
Comment (BugsNash) [reply js/controllers/window.js]: SGTM, done!

**Pull request merged**